docs/src/main/asciidoc/spring-cloud-gateway.adoc

```diff
@@ -1275,6 +1275,34 @@ spring:
 
 For some usages of the gateway, properties will be adequate, but some production use cases will benefit from loading configuration from an external source, such as a database. Future milestone versions will have `RouteDefinitionLocator` implementations based off of Spring Data Repositories such as: Redis, MongoDB and Cassandra.
 
+== Route metadata configuration
+Additional parameters can be configured for each route using metadata:
+
+.application.yml
+[source,yaml]
+----
+spring:
+  cloud:
+    gateway:
+      routes:
+      - id: route_with_metadata
+        uri: https://example.org
+        metadata:
+          optionName: "OptionValue"
+          compositeObject:
+            name: "value"
+          iAmNumber: 1
+----
+
+All metadata properties could be acquired from exchange:
+```
+Route route = exchange.getAttribute(GATEWAY_ROUTE_ATTR);
+// get all metadata properties
+route.getMetadata();
+// get a single metadata property
+route.getMetadataOption(someKey);
```

Comment:
`MetadataOption` needs to be renamed in docs.

Reply:
done